pkg/util/workloadspread/workloadspread.go

```diff
@@ -640,14 +645,21 @@ func injectWorkloadSpreadIntoPod(ws *appsv1alpha1.WorkloadSpread, pod *corev1.Po
 		cloneBytes, _ := json.Marshal(pod)
 		modified, err := strategicpatch.StrategicMergePatch(cloneBytes, subset.Patch.Raw, &corev1.Pod{})
 		if err != nil {
-			klog.ErrorS(err, "failed to merge patch raw", "raw", subset.Patch.Raw)
+			logger.Error(err, "failed to merge patch raw", "raw", subset.Patch.Raw)
 			return false, err
 		}
 		newPod := &corev1.Pod{}
 		if err = json.Unmarshal(modified, newPod); err != nil {
-			klog.ErrorS(err, "failed to unmarshal to Pod", "pod", modified)
+			logger.Error(err, "failed to unmarshal to Pod", "pod", modified)
 			return false, err
 		}
+		if newPod.Spec.PriorityClassName != pod.Spec.PriorityClassName {
+			// Workloadspread webhook is called after builtin k8s pod webhook, which means the priority
```

Comment:
consider rephase the comment as follows:

Workloadspread webhook is called after builtin admission plugin, which means the priority is already set before priorityClassName being patched. Mismatched priorityClassName and priority value will be rejected by apiserver. we have to clear priority to avoid the problem, and the builtin admission plugin will be reinvoked after kruise webhook to setting the correct priority value.